python/tvm/relay/transform/fake_quantization_to_integer.py

```diff
@@ -502,6 +502,33 @@ def register_binary_qnn(op_name, op):
 
     def binary(expr, type_map):
         left, right, left_t, right_t, out_t = get_binary_types(expr, type_map)
+
+        if (
+            approx_equal(left_t.scale, right_t.scale)
```

Comment:
Hmm I don't believe this work around is a good idea because it does not handle overflow well where qnn ops will. 

e.g. if we had 128 * 128 --> we will have issues.  

Hmm the main problem we have if I understand correctly is dealing with scales and zero points which are of shape [1] rather than be scalars. 

Can we just unpack scales and zero points into scalars then?

Reply:
`out_t.scale` and `out_t.zero_point` can be of shape `(channels,)` if the addition is used as bias add. If I comment out this if statement and run the test attached in this PR, I get

`AssertionError: The output scale needs to be a scalar, but got a tensor of shape (256,)` 

from the assert I added.

I think there is something weird in the implementation of FQ2I, since it doesn't make sense to have an output scale of shape `(channels,)`.  But that's how it works now, so this is the workaround I came up with.

Reply:
But right, this code path is only meant for catching 32 bit `add` used as bias addition. I'll update the condition to catch only this case.

Reply:
yes that is fine by me, can you update the comment to warn that dtype == 32 bits is really important for avoiding overflow.